app/controllers/workflow_credential_controller.rb

```diff
@@ -0,0 +1,122 @@
+class WorkflowCredentialController < ApplicationController
+  before_action :check_prototype
+  before_action :check_privileges
+  before_action :get_session_data
+
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  include Mixins::GenericListMixin
+  include Mixins::GenericSessionMixin
+  include Mixins::GenericShowMixin
+  include Mixins::BreadcrumbsMixin
+
+  menu_section :workflow_credentials
+
+  def self.display_methods
+    %w[repositories]
+  end
+
+  def self.model
+    ManageIQ::Providers::Workflows::AutomationManager::Credential
+  end
+
+  def display_repositories
+    nested_list(ManageIQ::Providers::Workflows::AutomationManager::ConfigurationScriptSource)
+  end
+
+  def show_searchbar?
+    true
+  end
+
+  def button
+    case params[:pressed]
+    when 'embedded_automation_manager_credentials_add'
+      javascript_redirect(:action => 'new')
+    when 'embedded_automation_manager_credentials_edit'
+      javascript_redirect(:action => 'edit', :id => params[:miq_grid_checks])
+    when 'ansible_credential_tag'
+      tag(self.class.model)
+    when "workflow_repository_tag" # repositories from nested list
+      tag(ManageIQ::Providers::Workflows::AutomationManager::ConfigurationScriptSource)
+    end
+  end
+
+  def new
+    assert_privileges('embedded_automation_manager_credentials_add')
+    drop_breadcrumb(:name => _("Add a new Credential"), :url => "/workflow_credential/new")
+    @in_a_form = true
+    @id = 'new'
+  end
+
+  def edit
+    assert_privileges('embedded_automation_manager_credentials_edit')
+    auth = ManageIQ::Providers::Workflows::AutomationManager::Authentication.find(params[:id].to_i)
```

Comment:
This class doesn't exist so it is resolving up a level
```
>> ManageIQ::Providers::Workflows::AutomationManager::Authentication
=> ManageIQ::Providers::EmbeddedAutomationManager::Authentication
```

I think you'd be better off using `self.class.model` here